spec/lita/handlers/stacker_spec.rb

```diff
@@ -1,4 +1,198 @@
+  it { is_expected.to route_command('unstack @user').to(:lifo_remove) }
+  it { is_expected.to route_command('stack drop').to(:lifo_remove) }
+  it { is_expected.to route_command('stack done').to(:lifo_remove) }
+  it { is_expected.to route_command('stack show').to(:lifo_peek) }
+  it { is_expected.to route_command('stacks show').to(:lifo_peek) }
+  it { is_expected.to route_command('stack clear').to(:lifo_clear) }
+  it { is_expected.to route_command('stacks clear').to(:lifo_clear) }
+
+  shared_examples 'a private chat' do
+    let(:command_options) { {} }
+
+    before do
+      send_command(command, command_options)
+    end
+
+    it 'does nothing' do
+      expect(replies).to be_empty
+    end
+  end
+
+  shared_context 'in a room' do
+    let(:command_options) { { from: Lita::Room.create_or_update('#public_channel') } }
+  end
+
+  describe '#lifo_add' do
+    let(:existing_stack) { [] }
+
+    before do
+      existing_stack.each { |s| send_command("stack @#{s}", command_options) }
+      send_command(command, command_options)
+    end
+
+    shared_examples 'adding users' do
+      it_behaves_like 'a private chat'
+
+      context 'when the message is in a room' do
+        include_context 'in a room'
+
+        it 'informs the user they have the floor' do
+          expect(replies.last).to include 'have the floor'
+        end
+
+        it 'adds the current user' do
+          send_command 'stack show', command_options
+          expect(replies.last).to include(target_user.name)
+        end
+
+        context 'when there is a stack' do
+          let(:existing_stack) { ['Trillian'] }
+
+          it 'informs the user who is before them' do
+            expect(replies.last).to include "after @#{existing_stack.last}"
+          end
+
+          it 'adds the user' do
+            send_command 'stack show', command_options
+            expect(replies.last).to include(target_user.mention_name)
+          end
+        end
+
+        context 'when the user is already in the stack' do
+          before do
+            send_command(command, command_options)
+          end
+
+          it 'informs the user' do
+            expect(replies.last).to include('already')
+          end
+
+          it 'does not add to the stack' do
+            send_command 'stack show', command_options
+            expect(replies.last.split(/\n/).grep(/\d+\. #{target_user.mention_name.tr('@', '')}/).size).to eq 1
+          end
+        end
+      end
+    end
+
+    context 'with an empty command' do
+      let(:command) { 'stack' }
+
+      include_examples 'adding users' do
+        let(:target_user) { user }
+      end
+    end
+
+    context 'with another user name' do
+      let(:target_user) { Lita::User.create(123, name: 'Zaphod') }
+      let(:command) { "stack @#{target_user.name}" }
+
+      include_examples 'adding users'
+    end
+
+    context 'with an on command' do
+      let(:command) { 'stack on that' } # perhaps this should jump to the top?
```

Comment:
might also be good to include what someone is stacking on in the `stack show` list in the far-flung future of potential updates

Reply:
Ooh, yes. Good idea. Though it should also have a special case for “that,” since that’s not useful. :)